staging/src/k8s.io/apiextensions-apiserver/test/integration/table_test.go

```diff
@@ -256,6 +257,48 @@ func TestTableGet(t *testing.T) {
 	}
 }
 
+func TestColumnsPatch(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, apiextensionsfeatures.CustomResourceWebhookConversion, true)()
+	tearDown, config, _, err := fixtures.StartDefaultServer(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tearDown()
+
+	apiExtensionClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// CRD with no top-level and per-version columns should be created successfully
+	crd := NewNoxuSubresourcesCRDs(apiextensionsv1beta1.NamespaceScoped)[0]
+	crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, dynamicClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// One should be able to patch the CRD to use per-version columns. The top-level columns
```

Comment:
add a patch test that updates a CRD that only has the defaulted top-level columns by explicitly setting the top-level columns to `null` and setting per-version columns instead?